framework/freedomotic-core/src/test/java/com/freedomotic/rules/StatementTest.java

```diff
@@ -0,0 +1,84 @@
+package com.freedomotic.rules;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.lang.reflect.InvocationTargetException;
+
+public class StatementTest {
+
+  @Test
+  public void createInput1() {
+    final Statement statement = new Statement();
+    final Statement retval = statement.create("!!!!", "01111", "!", "  !\"\"\"");
```

Comment:
The values are not good. 
`logical` should be taken from these constants: https://github.com/freedomotic/freedomotic/blob/master/framework/freedomotic-core/src/main/java/com/freedomotic/rules/Statement.java#L36 (and a test for a wrong one that is not in that list). `operand` is from the binary operators in this list, supposedly (it's not really enforced anywhere).
`attribute` should be identifier-like.
`value` an be any string (identifier-like, integers, etc are probably good choices).

Look at this class https://github.com/freedomotic/freedomotic/blob/be631f6045df67d89d6210b6a89e345983d83172/framework/freedomotic-core/src/main/java/com/freedomotic/rules/Payload.java
which uses `Statement`. Testing that class too would be a great thing to make the PR stronger.

Reply:
Test names are not good. The test names for tests that test the same method should point out the difference between these tests (typically the input value that changes resp. good vs bad inputs).